rewriter/Minitest.cc

```diff
@@ -420,4 +420,64 @@ vector<ast::ExpressionPtr> Minitest::run(core::MutableContext ctx, bool isClass,
     return stats;
 }
 
+// Move the contents of the `setup` method inside a synthetic `initialize` to allow non-nilable instance variables to be
+// declared
+//
+//    class Bar < Minitest::Test
+//      def setup
+//        @foo = T.let(42, Integer) # Error: The instance variable `@foo` must be declared
+//                                  # inside `initialize` or declared `nilable`
+//      end
+//    end
+//
+// One limitation is that this only works on immediate subclasses of Minitest::Test, given that at this stage we don't
+// really have more information about the class hierarchy.
+void rewriteMinitestSetupAsInitialize(core::MutableContext ctx, ast::ClassDef *klass) {
+    if (klass->kind != ast::ClassDef::Kind::Class || klass->ancestors.empty()) {
+        return;
+    }
+
+    auto *parent = ast::cast_tree<ast::UnresolvedConstantLit>(klass->ancestors.front());
+    if (parent == nullptr || parent->cnst != core::Names::Constants::Test()) {
+        return;
+    }
+
+    auto *scope = ast::cast_tree<ast::UnresolvedConstantLit>(parent->scope);
+    if (scope == nullptr || scope->cnst != core::Names::Constants::Minitest()) {
+        return;
+    }
+
+    std::vector<ast::ExpressionPtr> stats;
+    for (auto &stat : klass->rhs) {
+        if (auto *mdef = ast::cast_tree<ast::MethodDef>(stat)) {
+            if (mdef->name == core::Names::setup()) {
+                auto loc = mdef->loc;
+                auto method_name = core::Names::initialize();
+                auto arg = ast::MK::Local(loc, core::Names::name());
+                auto method = ast::MK::SyntheticMethod1(loc, loc, method_name, std::move(arg), std::move(mdef->rhs));
+
+                ast::Send::ARGS_store sigArgs;
+                sigArgs.emplace_back(ast::MK::Symbol(loc, core::Names::name()));
+                sigArgs.emplace_back(
+                    ast::MK::Send0(loc, ast::MK::Constant(loc, core::Symbols::T()), core::Names::untyped(), loc));
+
+                auto method_with_sig =
```

Comment:
Wouldn't that still break if someone defines an initializer with splats:

```rb
class MyTest < Minitest::Test
  extend T::Sig

  sig { params(arg: T.untyped).void }
  def initialize(*arg)
    super(arg)
  end
end
```

https://sorbet.run/#%23%20typed%3A%20strict%0A%0Aclass%20MyTest%20%3C%20Minitest%3A%3ATest%0A%20%20extend%20T%3A%3ASig%0A%0A%20%20sig%20%7B%20params%28arg%3A%20T.untyped%29.void%20%7D%0A%20%20def%20initialize%28*arg%29%0A%20%20%20%20super%28arg%29%0A%20%20end%0A%0A%20%20sig%20%7B%20params%28name%3A%20String%29.void%20%7D%0A%20%20def%20initialize%28name%29%0A%20%20%20%20super%28name%29%0A%20%20end%0Aend